samples/NativeLibrary/README.md

```diff
@@ -9,21 +9,21 @@ Create a .NET Core class library project using `dotnet new classlib -o NativeLib
 ## Building static libraries
 
 ```bash
-> dotnet publish /p:NativeLib=Static -r <RID> -c <Configuration>
+> dotnet publish /p:NativeLib=Static /p:SelfContained=true -r <RID> -c <Configuration>
```

Comment:
I was trying to figure out why this is not a problem when publishing an app as opposed to a library. It seems like it hits some defaults that set it to true. Could we just set this unconditionally in the targets instead? Would it hurt something else?

Reply:
The original problem is the error check here: https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L153

So we are going to suppress this error check by setting `SelfContained` to `true`. The problem with `SelfContained` property is that it controls NuGet restore behavior. We cannot set it for NuGet restore time since our package is not downloaded yet. And when `SelfContained` is not set at restore time and later gets set for build time, you will see errors like: `C:\Program Files\dotnet\sdk\5.0.102\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(387,5): error NETSDK1112: The runtime pack for Microsoft.NETCore.App.Runtime.win-x64 was not downloaded. Try running a NuGet restore with the RuntimeIdentifier 'win-x64'.`. It leads to never ending sequence of hacks.

I think the proper fix for this would be to fix the error check in Microsoft.NET.ILLink.targets that's causing the problem in the first place.

Reply:
Thanks for the explanation!